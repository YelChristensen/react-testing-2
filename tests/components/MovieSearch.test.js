import React from 'react';
import renderer from 'react-test-renderer';
import MovieSearch from '../../src/components/MovieSearch';
import { shallow } from 'enzyme';




describe('MovieSearch', () => {
  test('matches the snapshot', () => {
    const tree = renderer.create(<MovieSearch />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('handleSumbit gets called when the form is submitted', () => {
    const event = {
        preventDefault: jest.fn()
    }
    const handleSumbit = jest.fn()
    const wrapper = shallow(
        <MovieSearch 
            movieString={'Gone with the wind'} 
            submitSearch={handleSumbit} 
        />)
    wrapper.find('form').simulate('submit', event);
    expect(handleSumbit.mock.calls).toEqual([['Gone with the wind']])
})
});

