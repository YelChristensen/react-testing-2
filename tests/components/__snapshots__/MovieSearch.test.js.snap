// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearch matches the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="movie-search"
  >
    Search for a movie
  </label>
  <input
    onChange={[Function]}
    type="text"
    value={undefined}
  />
</form>
`;
